etna/insights/models.py

```diff
@@ -46,12 +59,29 @@ class InsightsPage(HeroImageMixin, TeaserImageMixin, BasePage):
 
     sub_heading = models.CharField(max_length=200, blank=False)
     body = StreamField(InsightsPageStreamBlock, blank=True, null=True)
-
+    topic = models.ForeignKey(
+        "collections.TopicExplorerPage",
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name="+",
+    )
+    time = models.ForeignKey(
+        "collections.TimePeriodExplorerPage",
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name="+",
+    )
+    tags = ClusterTaggableManager(through=InsightsPageTag, blank=True)
```

Comment:
As mentioned in the dev notes: 
> The tag value will need to be indexed in order to make use of Wagtail’s search() feature for identifying ‘similar’ pages to display on each page. 

Basically, this means you'll need to add something like this:

```
    def get_insight_tag_names(self):
        return '\n'.join(self.tags.all().values_list('name', flat=True))

    search_fields = Page.search_fields + [
        index.SearchField('get_insight_tag_names'),
    ]
```

Here is the [relevant section of the docs](https://docs.wagtail.org/en/stable/topics/search/indexing.html#indexing-extra-fields).

Reply:
Yes, thank you, done now.